### PR TITLE
Added annotation to indicate tests should run in UI thread.

### PR DESCRIPTION
**Pull request by cafesilencio**
### Description
Fix for #6035 by adding @UiThreadTest to get the tests running on the UI thread,

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Mapbox welcomes participation and contributions from everyone.
 
 ## Unreleased
 #### Features
+- Added `MapboxNavigation.requestRoadGraphDataUpdate` method and `RoadGraphDataUpdateCallback` class to request road graph data update and recreate `MapboxNavigation` instance in the callback to be able to use latest data even if the application lifecycle is very long. [#6044](https://github.com/mapbox/mapbox-navigation-android/pull/6044)
```
Comment (dzinad) [inline CHANGELOG.md:7]: Why is it here? And pretty much everything related to `requestRoadGraphDataUpdate`.

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/RouteLineLayersTest.kt [hunk 2]
```diff
@@ -54,6 +55,7 @@ class RouteLineLayersTest : BaseTest<BasicNavigationViewActivity>(
     }
 
     @Test
+    @UiThreadTest
```
Comment (dzinad) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/RouteLineLayersTest.kt:58]: All interactions with `MapboxNavigation` are required to be done from the main thread. It's just that only in some cases the exception is thrown. Shouldn't we approach this task systematically? For example. run all tests on main thread if it's possible. We use `MapboxNavigation` pretty much everywhere, who knows which tests will suddenly start failing tomorrow or in a month.

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/RouteLineLayersTest.kt [hunk 3]
```diff
@@ -166,6 +168,7 @@ class RouteLineLayersTest : BaseTest<BasicNavigationViewActivity>(
     // for the primary route should be moved to the top of the stack above the other
     // route line layers.
     @Test
+    @UiThreadTest
```
Comment (LukasPaczos) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/RouteLineLayersTest.kt:171]: We cannot run the whole test on the main thread. The test includes a latch so as soon as we reach `countDownLatch.await()` I would imagine that the main thread deadlocks and the test should would just time out.

Instead, we should move the contents that need to be run on the main thread (interactions with the SDK) into a `runOnMainSync` block, like:
```
@Test
fun test() {
  val latch: Latch
  runOnMainSync {
    // interact with SDK and count down
  }
  latch.await()
}
```

There are many examples of that across other tests.

Comment (cafesilencio) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/RouteLineLayersTest.kt]: I forgot about this.

**Review by dzinad — APPROVED** [review]

**Pull request merged**